src/mixins/metadata.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { addMetadata } from 'docc-render/utils/metadata';
+
+export default {
+  methods: {
+    /**
+     * Extract and join all the values from key 'text' on every object inside an array of objects,
+     * if there is an inlineContent array, it will extract all its text recursively as well
+     * @param {Array} content - array of objects containing "text" key
+     * @returns {String}
+     */
+    extractText(content) {
+      return content.reduce((acc, { text, inlineContent = [] }) => (
+        acc.concat(text || this.extractText(inlineContent)).filter(Boolean)
+      ), []).join('');
+    },
+  },
+  computed: {
+    pagePath: ({ $route: { path = '/' } = {} }) => path,
+  },
+  created() {
+    addMetadata({
```

Comment:
Maybe `addOrUpdateMetadata` might be a better name? We should also make sure that we aren't accidentally duplicating meta tags for the scenario where a user is transitioning between pages—it would be good to manually test the scenario where we go from a doc page to a tutorial (or vice versa) and make sure that we are only updating the things that are expected to change dynamically from page to page (maybe only the description?).

Reply:
I updated the name: https://github.com/apple/swift-docc-render/pull/46/commits/d1b2f4ccad01ca1e2f5dd3e3dbffec618dafb321

Because I moved the static images to the HTML, now we are only updating the data that changes from one page to another: title, description and path. These params should always be different from one page to another, so we are good. Also I made sure to delete descriptions if page doesn't have so we don't inherit from previous page.

Reply:
Ah, I see that you changed the name, but we need to also change the logic to handle the update case. When I click a link to a tutorial page from a documentation page, I'm observing that there are duplicated `<meta>` tags for both pages on the second page.

We'll need to update the logic for creating these nodes to first check if they exist and update them instead of creating/adding them in that scenario.

Reply:
Done! https://github.com/apple/swift-docc-render/pull/46/commits/731200820b793e12030a9566b072973230bf09c7